pandas/core/strings/accessor.py

```diff
@@ -243,6 +243,9 @@ def __getitem__(self, key):
         result = self._data.array._str_getitem(key)
         return self._wrap_result(result)
 
+    def __iter__(self):
```

Comment:
` __iter__: ClassVar[None]  # type: ignore[assignment]` could potentially help type checkers (I didn't try it) or using `def __iter__(self) -> NoReturn:`

Reply:
While this passes the `is_list_like` test, this appears not to completely address `iter(ser.str)` raising because `iter` still works because `__getitem__` is defined